examples/composite_dialog/my_app_window/mod.rs

```diff
@@ -0,0 +1,80 @@
+mod imp;
+
+use gtk::{
+    glib::{self, ParamSpec},
+    prelude::*,
+    ResponseType,
+};
+
+glib::wrapper! {
+    pub struct MyAppWindow(ObjectSubclass<imp::MyAppWindow>)
+        @extends gtk::Widget, gtk::Window, gtk::ApplicationWindow;
+}
+
+#[gtk::template_callbacks]
+impl MyAppWindow {
+    pub fn new<P: glib::IsA<gtk::Application>>(app: &P) -> Self {
+        glib::Object::builder().property("application", app).build()
+    }
+
+    /// Callback handler for notify::label signal.
+    ///
+    /// When counter property reach 3, a dialog pops up and present the user
+    /// with 2 choices: Set the counter to 6 or reset the counter to 0.
+    #[allow(deprecated)]
+    #[template_callback]
+    fn popup_dialog(&self, _p: &ParamSpec) {
+        // Check counter property and create a Dialog.
+        if self.counter() == 3 {
+            let dial = gtk::Dialog::with_buttons(
+                Some("Counter value is 3"),
+                Some(self),
+                gtk::DialogFlags::MODAL,
+                &[
+                    ("Set counter to 6", ResponseType::Other(35)),
+                    ("Reset counter", ResponseType::Ok),
+                ],
+            );
+            dial.set_transient_for(Some(self));
+
+            let app = self.clone();
+
+            // Closure handling response signal from gtk::Dialog.
+            // The signature of the function differ from the documentation for response signal.
+            // gtk-rs use an i32 instead of a [`gtk::ResponseType`] as a response signal.
+            dial.connect_closure(
```

Comment:
There is already a connect_response that does the type conversion for you. The intention of the example is to create the dialog in the UI file and connecting to the response signal from it so that you would get a crash if you try to use ResponseType directly in the template callback. The example has to be adjusted to show that

Reply:
I added _gtk::Dialog_ in ui file and manage its response signal with a handler. A short comment was added to highlight the presence of type _i32_ instead of _ResponseType_ in the callback.